packages/pds/src/scripts/common.ts

```diff
@@ -0,0 +1,27 @@
+import { Keypair } from '@atproto/crypto'
+import AppContext from '../context'
+
+export const updatePlcSigningKey = async (
+  ctx: AppContext,
+  did: string,
+  signingKey?: Keypair,
+) => {
+  const updateTo = signingKey ?? (await ctx.actorStore.keypair(did))
+  const currSigningKey = await ctx.idResolver.did.resolveAtprotoKey(did, true)
+  if (updateTo.did() === currSigningKey) {
+    // already up to date
+    return
+  }
+  if (ctx.entrywayAdminAgent) {
+    await ctx.entrywayAdminAgent.api.com.atproto.admin.updateAccountSigningKey({
+      did,
+      signingKey: updateTo.did(),
+    })
```

Comment:
I think we'll always land in this case since `entrywayAdminAgent` exists when there's an entryway admin token, which fallsback to `env.adminPassword`, which should always be set.

Reply:
it's also contingent on having an entryway cfg as well which is the main thing